### PR TITLE
Delete Logs depends on user defined values

### PR DESCRIPTION
**Pull request by ohmcodes**
- MaxFiles default 20
- MaxAgeInSeconds default 86400 = 24hrs
- EnableDeleteLogging default true : Show deleted files in console

### CODE REVIEW

File: AsaApi/main.cpp [hunk 1]
```diff
@@ -63,6 +63,67 @@ void OpenConsole()
 	SetConsoleOutputCP(CP_UTF8);
 }
 
+void DeleteOldLogFiles()
```
Comment (SubstituteR) [inline AsaApi/main.cpp:66]: We should probably enable some form of log file name configuration in the long term.

File: AsaApi/main.cpp [hunk 2]
```diff
@@ -63,6 +63,67 @@ void OpenConsole()
...
+	const bool enableDeleteLogging = config["settings"].value("EnableDeleteLogging", true);
+
+	std::vector<std::pair<std::string, time_t>> fileTimePairs;
+	// Iterate through the files in the folder
+	for (const auto& entry : std::filesystem::directory_iterator(folderPath)) {
+		if (entry.is_regular_file()) {
+			std::string fileName = entry.path().filename().string();
+
+			// Extract datetime from the file name
+			std::string dateTimeString = fileName.substr(14, 19); // Extract data and time from "ArkApi_00000_yyyy-mm-dd_hh-mm.log"
```
Comment (SubstituteR) [inline AsaApi/main.cpp:89]: I'm not a fan of this, we should probably consider the actual system date/time of the file.

File: AsaApi/main.cpp [hunk 3]
```diff
@@ -63,6 +63,67 @@ void OpenConsole()
...
+
+			struct tm tm = {};
+			std::istringstream ss(dateTimeString);
+			ss >> std::get_time(&tm, "%Y-%m-%d_%H-%M");
+
+			if (ss.fail()) {
+				Log::GetLog()->error("Failed to parse datetime from file: {}", fileName);
```
Comment (SubstituteR) [inline AsaApi/main.cpp:96]: I think this should be either an info or a debug log.

File: AsaApi/main.cpp [hunk 4]
```diff
@@ -63,6 +63,67 @@ void OpenConsole()
...
+
+			// Convert tm to time_t
+			time_t fileTime = std::mktime(&tm);
+
+			fileTimePairs.push_back(std::make_pair(fileName, fileTime));
+		}
+	}
+
+	// Sort files by time in descending order
+	std::sort(fileTimePairs.begin(), fileTimePairs.end(),
```
Comment (SubstituteR) [inline AsaApi/main.cpp:108]: Ultimately this pre-optimization is going to have more overhead than just iterating the list of the files and checking their dates directly has the same time complexity as the sort.

File: Configs/config.json [hunk 5]
```diff
@@ -1,8 +1,11 @@
...
+{
+  "settings": {
+    "AutomaticPluginReloading": true,
+    "AutomaticPluginReloadSeconds": 5,
     "SaveWorldBeforePluginReload": true,
-    "AttachToParent": true
+    "AttachToParent": true,
+    "MaxFiles": 20,
+    "MaxAgeInSeconds": 86400,
+    "EnableDeleteLogging": true
```
Comment (SubstituteR) [inline Configs/config.json:9]: I don't think we need this.

We can use an info/audit log instead and the end-user can set their log level appropriately.

**Review by SubstituteR — CHANGES_REQUESTED** [review]
1. Should log rotation be handled by the loader or the API?
2. If the Loader rotates logs, should it pass along log configuration options to the API?
3. Should logging be more configurable wrt naming schemes?

File: AsaApi/main.cpp (hunk 2 above)
Comment (ohmcodes) [reply AsaApi/main.cpp:89]: i found a bug using substr as the process id could be 2 digits i will change it maybe splitting those underscore to get the date/time and i will look into reading the actual write date of the file

File: Configs/config.json (hunk 5 above)
Comment (ohmcodes) [reply Configs/config.json:9]: Every user has their preferences how they want to keep their logs, for me while developing a plugin this logs can be large in count as a turning on and off the server more often in a day i can have like more than 200 files though its not that big but imaging a week of server have their daily restart or some sort, if you are referring to the log `EnableDeleteLogging` well that can be remove

Comment (SubstituteR) [reply Configs/config.json:9]: Yes, I was referring to `EnableDeleteLogging`.

**Pull request closed without merging**